core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -203,18 +203,10 @@ public SearchSourceBuilder(StreamInput in) throws IOException {
         postQueryBuilder = in.readOptionalNamedWriteable(QueryBuilder.class);
         queryBuilder = in.readOptionalNamedWriteable(QueryBuilder.class);
         if (in.readBoolean()) {
```

Comment:
Can empty list be a valid value here so we can avoid the boolean? Not needed as part of this PR but it'd be nice.